include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -96,28 +94,25 @@ single_buffer_to_vectors(typename std::vector<T>::iterator begin_buffer,
                          const std::vector<size_t>& dims,
                          const size_t current_dim,
                          std::vector<T>& vec_single_dim) {
-    const size_t n_elems = dims[current_dim];
-    typename std::vector<T>::iterator end_copy_iter =
-        std::min(begin_buffer + static_cast<long>(n_elems), end_buffer);
+    const long n_elems = static_cast<long>(dims[current_dim]);
```

Comment:
Remove duplicate type information:

```suggestion
    const auto n_elems = static_cast<long>(dims[current_dim]);
```